src/Metal/MTLDevice.cs

```diff
@@ -42,6 +60,28 @@ public static partial class MTLDevice {
 				return system_default;
 			}
 		}
+
+#if MONOMAC
+		[Mac (10,13, onlyOn64: true), NoiOS, NoWatch, NoTV]
+		[DllImport (Constants.MetalLibrary)]
+		static extern IMTLDevice[] MTLCopyAllDevicesWithObserver (IntPtr observer, MTLDeviceNotificationHandler handler);
+
+		[Mac (10,13, onlyOn64: true), NoiOS, NoWatch, NoTV]
+		public static IMTLDevice [] GetAllDevices (MTLDeviceNotificationHandler handler, ref NSObject observer)
```

Comment:
Then you've defined the P/Invoke incorrectly:

```
static extern IMTLDevice[] MTLCopyAllDevicesWithObserver (ref IntPtr observer, MTLDeviceNotificationHandler handler);
```

and you're calling it incorrectly:

```
IntPtr h = observer.Handle;
var rv = MTLCopyAllDevicesWithObserver (ref h, handler);
var obj = Runtime.GetNSObject (rv);
if (h != observer.Handle)
    observer = Runtime.GetNSObject (h);
return obj;
```